packages/itwin/tree-widget/src/components/trees/stateless/common/components/BaseTree.tsx

```diff
@@ -134,6 +125,12 @@ function BaseTreeRenderer({
   const reportingExpandNode = useReportingAction({ action: expandNode, reportUsage });
   const reportingOnFilterClicked = useReportingAction({ action: onFilterClick, reportUsage });
 
+  useEffect(() => {
+    if (rootNodes?.length && !isLoading) {
+      onRootNodesLoaded?.();
+    }
+  }, [onRootNodesLoaded, isLoading, rootNodes]);
```

Comment:
I think this can be done in `onPerformanceMeasured` callback passed to `useUnifiedSelectionTree`. Also it looks like that current implementation invokes `onRootNodesLoaded` every time any hierarchy level is loaded.

Reply:
Moved this logic to `onPerformanceMeasured` when `action === "reload"`